src/main.ts

```diff
@@ -0,0 +1,528 @@
+      execution_env.NIX_INSTALLER_PROXY = this.proxy;
+    }
+
+    if (this.ssl_cert_file !== null) {
+      execution_env.NIX_INSTALLER_SSL_CERT_FILE = this.ssl_cert_file;
+    }
+
+    if (this.diagnostic_endpoint !== null) {
+      execution_env.NIX_INSTALLER_DIAGNOSTIC_ENDPOINT =
+        this.diagnostic_endpoint;
+    }
+
+    // TODO: Error if the user uses these on not-MacOS
+    if (this.mac_encrypt !== null) {
+      execution_env.NIX_INSTALLER_ENCRYPT = this.mac_encrypt;
+    }
+
+    if (this.mac_case_sensitive !== null) {
+      execution_env.NIX_INSTALLER_CASE_SENSITIVE = this.mac_case_sensitive;
+    }
+
+    if (this.mac_volume_label !== null) {
+      execution_env.NIX_INSTALLER_VOLUME_LABEL = this.mac_volume_label;
+    }
+
+    if (this.mac_root_disk !== null) {
+      execution_env.NIX_INSTALLER_ROOT_DISK = this.mac_root_disk;
+    }
+
+    if (this.logger !== null) {
+      execution_env.NIX_INSTALLER_LOGGER = this.logger;
+    }
+
+    if (this.log_directives !== null) {
+      execution_env.NIX_INSTALLER_LOG_DIRECTIVES = this.log_directives;
+    }
+
+    // TODO: Error if the user uses these on MacOS
+    if (this.init !== null) {
+      execution_env.NIX_INSTALLER_INIT = this.init;
+    }
+
+    if (this.start_daemon !== null) {
+      if (this.start_daemon) {
+        execution_env.NIX_INSTALLER_START_DAEMON = "true";
+      } else {
+        execution_env.NIX_INSTALLER_START_DAEMON = "false";
+      }
+    }
+
+    let extra_conf = "";
+    if (this.github_token !== null) {
+      extra_conf += `access-tokens = github.com=${this.github_token}`;
+      extra_conf += "\n";
+    }
+    if (this.trust_runner_user !== null) {
+      // TODO: Consider how to improve this
+      extra_conf += `trusted-users = root ${process.env.USER}`;
+      extra_conf += "\n";
+    }
+    if (this.extra_conf !== null && this.extra_conf.length !== 0) {
+      extra_conf += this.extra_conf.join("\n");
+      extra_conf += "\n";
+    }
+    execution_env.NIX_INSTALLER_EXTRA_CONF = extra_conf;
+
+    if (process.env.ACT && !process.env.NOT_ACT) {
+      actions_core.info(
+        "Detected `$ACT` environment, assuming this is a https://github.com/nektos/act created container, set `NOT_ACT=true` to override this. This will change the settings of the `init` as well as `extra-conf` to be compatible with `act`",
+      );
+      execution_env.NIX_INSTALLER_INIT = "none";
+    }
+
+    return execution_env;
+  }
+
+  private async execute_install(binary_path: string): Promise<number> {
+    const execution_env = this.executionEnvironment();
+    actions_core.info(
+      `Execution environment: ${JSON.stringify(execution_env, null, 4)}`,
+    );
+
+    const args = ["install"];
+    if (this.planner) {
+      args.push(this.planner);
+    } else {
+      args.push(get_default_planner());
+    }
+
+    if (this.extra_args) {
+      const extra_args = stringArgv(this.extra_args);
+      args.concat(extra_args);
+    }
+
+    const merged_env = {
```

Comment:
I wonder how this behaves in the case where a user specifies `env: NIX_INSTALLER_EXTRA_CONF: "asdf"` and `with: extra-conf: "foo"`... Will it choose the input over the env, or error because there are two envs with the same name but conflicting values?

Reply:
In theory a user could not modify the env provided to the javascript!

Reply:
Ahhh ya, fair.